src/main/java/com/google/sps/servlets/CommentAnalysis.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+
+// This class encapsulates the each element in json comment array into separate user comment object
+// and do sentiment analysis on each of them
+public class CommentAnalysis {
+  private UserComment comment;
+
+  public CommentAnalysis(CommentThread originalComment) {
```

Comment:
I am honestly somewhat confused by the implementation of this class. Since the API returns a list of comment threads, a new instance of this class would need to be instantiated up to 100 times since each seems to be attached to the UserComment. At a glance, it looks like it'll be implemented like: "new CommentAnalysis(comment).sentiAnalysis()" to return the sentiment score for a comment. However, if that was your intention, would it not be better to give it a static method that takes in a userComment then returns the sentiment analysis score: "CommentAnalysis.sentiAnalysis(comment)". Since converting from the raw CommentThread to the userComment is a single line of code, it could even be done on the fly within the sentiAnalysis() method.

Reply:
+1